enhancements/removing-midstream.md

```diff
@@ -0,0 +1,322 @@
+```
+
+### Read operator metadata and fetch payload versions
+
+For this to work, we need to assume the upstream `tektoncd/operator` repository (or any personal fork)
+hold the payload information somewhere. As of today this is the case in [`test/config.sh`](https://github.com/tektoncd/operator/blob/main/test/config.sh),
+but this is subject to change if we feel like it.
+
+Given that assumption, the goal here would be to fetch the payload from this information and copy the
+content into the correct set of containers.
+
+```bash
+$ make sources/operator/fetch-payload
+# […]
+```
+
+TBD
+
+### Apply patches if need be
+
+All our container builds are done in multi-stage : one the builds and one that is the final image.
+We can use this to our advantage to apply the patches at build time, just after getting the sources in.
+In a nutshell we are looking to do the following (for each container image):
+
+- copy remote source
+- copy patches
+- apply patches
+- build
+
+This means, however that patches needs to live at the same level as the `Dockerfile`. As there is multiple 
+images for a given component, we will have to "duplicate" the `patches` on all component's.
+
+```bash 
+$ make patches
+# sync patches from ./patches to each container images
+$ make container/openshift-pipelines-controller/patches
+# sync patches for the openshift-pipelines-controller image
+```
+
+
+### Build images
+
+Images will be build by `brew` in the Red Hat Toolchain. That said, it should be possible to build images,
+and even the bundle, ahead of time. This is already the case, by doing a similar job as what CPaaS does with
+the templates (`.in` files).
+
+```bash 
+$ make images
+# builds all images but the operator bundle
+$ make images/push
+# builds and push all images but the operator bundle
+$ make images/digest
+# build, push and write image digest for all images but the operator bundle
+$ make container/openshift-pipelines-controller
+# build openshift-pipelines-controller image
+$ make container/openshift-pipelines-controller/push
+# build and push openshift-pipelines-controller image
+# […]
+```
+
+### Create a gerrit changeset
+
+- We need to setup some credentials (or use a bot account) to create a changeset for gerrit.
+  - **todo** know who to ping to create a bot for gerrit
+- Create a branch for the sync to happen (`upgrade-$(date …)`)
+- Update upstream source, sync patches and stuff
+- (optional) Build images to make sure it works
+- `git review -f {branch-name}` to create a changeset
+
+### Automation & execution
+
+This "process" should be run automatically at a given pace. Today the `poll-upstream` job runs daily, and will
+upgrade the gerrit repository daily, generating a daily build. 
+
+TBD
+
+### Where should this live
+
+One question is where all this automation should live. As of today, the choice is to make it live in the gerrit
+repository itself, making it the source of truth of mostly anything.
```

Comment:
i like this. This will also imply that the gerrit repo OpenShift pipelines will be the true ïntegrated product home", the one place where both configuration and automation live together.

Reply:
That's the idea. The only downside I see with it, is that any change we do, even if not on sources, patches, … will generate a build. I fix a type in a comment of a script, it will generate a build. It's not that bad but it might take resources from brew/osbs for no valid reason. But.. we can decide later to move if we feel like it.